apt/method_test.go

```diff
@@ -272,17 +281,18 @@ func TestAptMethodRunFail(t *testing.T) {
 	writer := MessageWriter{writer: stdinwriter}
 	writer.WriteMessage(Message{
 		code:        700,
-		description: "Bogus method",
+		description: "Malformed message",
+		fields:      map[string][]string{"": {"foo"}},
 	})
 
-	msg, err = reader.ReadMessage(ctx)
-	if err != nil {
-		t.Fatalf("failed, %v", err)
+	// If we receive a malformed message from `apt`, we immediately bail
+	// without sending an error response. Unlike the other tests, there is no
+	// protocol message to read here, but we can assert on the return value of
+	// `Run`.
+	runErr := <-errChan
```

Comment:
Please check the message of `runErr` to make sure it's the error you expect. (That is, after the nil check, add another check that runErr's message is "Malformed message" as expected.)